mesmerize_viz/_store_model.py

```diff
@@ -0,0 +1,158 @@
+        subscriber: fastplotlib.ImageGraphic, fastplotlib.LinearSelector, ipywidgets.IntSlider, or ipywidgets.FloatSlider
+            ipywidget or fastplotlib object to be synchronized
+        data: pynapple.TsdFrame, optional
+            If subscriber is a fastplotlib.ImageGraphic, must have an associating pynapple.TsdFrame to update data with.
+        data_filter: callable, optional
+            Function to apply to data before updating. Must return data in the same shape as input.
+        multiplier: int | float, optional
+            Scale the current time to reflect differing timescale.
+        """
+        # create a TimeStoreComponent
+        component = TimeStoreComponent(subscriber=subscriber,
+                                       data=data,
+                                       data_filter=data_filter,
+                                       multiplier=multiplier)
+
+        # add component to the store
+        self._store.append(component)
+
+        # add event handler to component.subscriber to call update_store
+        if isinstance(component.subscriber, (IntSlider, FloatSlider)):
+            component.subscriber.observe(self._update_store, "value")
+        if isinstance(component.subscriber, LinearSelector):
+            component.subscriber.add_event_handler(self._update_store, "selection")
+
+    def unsubscribe(self, subscriber: ImageGraphic | LinearSelector | IntSlider | FloatSlider):
+        """Remove a subscriber from the store."""
+        for component in self.store:
+            if component.subscriber == subscriber:
+                #  remove the component from the store
+                self.store.remove(component)
+                # remove event handler
+                if isinstance(component, (IntSlider, FloatSlider)):
+                    component.unobserve(self._update_store)
+                if isinstance(component, LinearSelector):
+                    component.subscriber.remove_event_handler(self._update_store, "selection")
+
+    def _update_store(self, ev):
+        """Called when event occurs and store needs to be updated."""
+        # parse event to see if it originated from ipywidget or selector
+        if isinstance(ev, FeatureEvent):
+            # check for multiplier to adjust time
+            for component in self.store:
+                if isinstance(component.subscriber, LinearSelector):
+                    if ev.graphic == component.subscriber:
+                        self.time = ev.info["value"] / component.multiplier
+        else:
+            self.time = ev["new"]
+
+        for component in self.store:
+            # update ImageGraphic data no matter what
+            if isinstance(component.subscriber, ScatterGraphic):
+                component.subscriber.data = component.data.get(self.time)
+            elif isinstance(component.subscriber, ImageGraphic):
+                if component.data_filter is None:
+                    new_data = component.data.get(self.time)
```

Comment:
all the `data.get()` calls will have to be replaced with regular numpy interger indexing where `time` is frame index, for example: `new_dat a= component.data[self.time]`

the `.get()` method of pynapple arrays is basically a getter based on `float` time.